### DISCUSSION TITLE
Add xxHash library function to DuckDB

### DISCUSSION THREAD

**Discussion by ashish01** [duckdb/duckdb#6228] [Ideas]
DuckDB current support md5 as a hashing mechanism. MD5 has good hash qualities as per smhasher test suite but it is slower compared to other well distributed hashes like xxhash. According to (https://github.com/Cyan4973/xxHash) slowest version of xxhash is 16x faster (0.6GB/s vs 9.7GB/s) than md5. I also did a simple benchmark on small random strings locally and XXH3_64 is atleast 6x faster than MD5 on my laptop.

```
Unable to determine clock rate from sysctl: hw.cpufrequency: No such file or directory
This does not affect benchmark measurements, only the metadata output.
2023-02-12T09:31:06-08:00
Running ./benchmark_xxhash
Run on (10 X 24.121 MHz CPU s)
CPU Caches:
  L1 Data 64 KiB
  L1 Instruction 128 KiB
  L2 Unified 4096 KiB (x10)
Load Average: 2.18, 2.38, 2.46
---------------------------------------------------------
Benchmark               Time             CPU   Iterations
---------------------------------------------------------
BM_XXH32             28.6 ns         28.6 ns     24242760
BM_XXH3_64bits       24.9 ns         24.9 ns     28078620
BM_MD5                161 ns          161 ns      4334741
```

xxhash is BSD licensed and comes as a single .h file.

**Comment by lnkuiper** [comment]
Hi @ashish01 thanks for the suggestion. I made a draft for a "hash" extension for DuckDB to which a bunch of popular hash functions could be added, like xxHash, cityhash murmurhash, etc. I will finish the extension and release it at some point so more hash functions are usable within DuckDB.

**Reply by PhE** [reply]
Shall we use those functions to get a hash of a row ? a table ? a database ? See [this discussion](https://github.com/duckdb/duckdb/discussions/6296)

**Reply by lnkuiper** [reply]
It would be a scalar function, which means you can use it to hash a single value. This value can be any type, e.g., `VARCHAR`, but also a row (`STRUCT`). The link you sent points back to this discussion, so I have seen it.

**Reply by PhE** [reply]
Fixed [discussions/6296 link](https://github.com/duckdb/duckdb/discussions/6296)
